crates/pallets/starknet/src/tests/declare_tx.rs

```diff
@@ -119,3 +98,58 @@ fn given_contract_declare_tx_fails_wrong_tx_version() {
         assert_err!(Starknet::declare(none_origin, transaction), Error::<Test>::TransactionExecutionFailed);
     })
 }
+
+#[test]
+fn given_contract_declare_on_openzeppelin_account_then_it_works() {
+    new_test_ext().execute_with(|| {
+        System::set_block_number(0);
+        run_to_block(2);
+        let none_origin = RuntimeOrigin::none();
+
+        let account_addr = get_account_address(AccountType::Openzeppelin);
+
+        let erc20_class = ContractClassWrapper::try_from(get_contract_class(ERC20_CONTRACT_PATH)).unwrap();
+        let erc20_class_hash =
+            <[u8; 32]>::from_hex("057eca87f4b19852cfd4551cf4706ababc6251a8781733a0a11cf8e94211da95").unwrap();
+
+        let tx_hash = H256::from_str("0x05aadce02f11b18562f0f2a519a34ca53e8f939b9bc829a4ef2a2ec7690a2c7b").unwrap();
+        let transaction = DeclareTransaction {
+            sender_address: account_addr,
+            contract_class: erc20_class,
+            version: 1,
+            compiled_class_hash: erc20_class_hash,
+            nonce: U256::zero(),
+            max_fee: U256::from(u128::MAX),
+            signature: sign_message_hash(tx_hash),
+        };
+
+        assert_ok!(Starknet::declare(none_origin, transaction));
```

Comment:
also check that the class is indeed declared and the declared class = the one that you declared. Something like that
```rs
assert_eq!(Starknet::contract_class_from_class_hash(erc20_class_hash), erc20_class)
```